### PR TITLE
OHRM5X-111: Termination Record FE & BE

### PR DESCRIPTION
**Pull request by ireshe**
need to add Delete warning when deleting the Already in use Termination Record

### CODE REVIEW

File: symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    /**
+     * @param $idArray
+     * @return bool
+     * @throws NonUniqueResultException
+     */
+    public function isReasonInUse($idArray): bool
+    {
+        $query = $this->createQueryBuilder(Employee::class, 'e');
+        $query->leftJoin('e.EmployeeTerminationRecord', 'et');
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php:177]: wrong mapping, ==> `e.employeeTerminationRecord `

File: symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        $query->leftJoin('et.TerminationReason', 'tr')
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php:178]: ==> `et.terminationReason`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            return false;
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param $idArray
+     * @return bool
+     * @throws NonUniqueResultException
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php:172]: will remove this

File: symfony/plugins/orangehrmPimPlugin/Service/TerminationReasonConfigurationService.php [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    public function isExistingTerminationReasonName(string $terminationReasonName): bool
+    {
+        return $this->getTerminationReasonDao()->isExistingTerminationReasonName($terminationReasonName);
+    }
+
+    /**
+     * @param $idArray
+     * @return bool
+     */
+    public function isReasonInUse($idArray): bool
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Service/TerminationReasonConfigurationService.php:129]:
```php
public function isReasonInUse(array $idArray): bool
```

File: symfony/plugins/orangehrmPimPlugin/config/routes.yaml [hunk 5]
```diff
@@ -271,6 +271,20 @@ apiv2_pim_employees_license:
     empNumber: '\d+'
     id: '\d+'
 
+apiv2_pim_termination_record:
+  path: /api/v2/pim/termination_records/{id}
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/config/routes.yaml:275]: `/api/v2/pim/termination-reasons/{id}`

File: symfony/plugins/orangehrmPimPlugin/config/routes.yaml [hunk 6]
```diff
@@ -271,6 +271,20 @@ apiv2_pim_employees_license:
...
+  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
+  methods: [ GET, PUT ]
+  defaults:
+    _api: OrangeHRM\Pim\Api\TerminationReasonConfigurationAPI
+
+apiv2_pim_termination_records:
+  path: /api/v2/pim/termination_records
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/config/routes.yaml:282]: `/api/v2/pim/termination-reasons`

File: symfony/plugins/orangehrmPimPlugin/config/routes.yaml [hunk 7]
```diff
@@ -271,6 +271,20 @@ apiv2_pim_employees_license:
     empNumber: '\d+'
     id: '\d+'
 
+apiv2_pim_termination_record:
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/config/routes.yaml:274]: `apiv2_pim_termination_reason`

File: symfony/plugins/orangehrmPimPlugin/config/routes.yaml [hunk 8]
```diff
@@ -271,6 +271,20 @@ apiv2_pim_employees_license:
...
+  path: /api/v2/pim/termination_records/{id}
+  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
+  methods: [ GET, PUT ]
+  defaults:
+    _api: OrangeHRM\Pim\Api\TerminationReasonConfigurationAPI
+
+apiv2_pim_termination_records:
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/config/routes.yaml:281]: `apiv2_pim_termination_reasons`

File: symfony/client/src/orangehrmPimPlugin/pages/configuration/EditTerminationReason.vue [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6">Edit Termination Reason</oxd-text>
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmPimPlugin/pages/configuration/EditTerminationReason.vue:24]: Please use class `orangehrm-main-title` for headings.

File: symfony/client/src/orangehrmPimPlugin/pages/configuration/SaveTerminationReason.vue [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6">Add Termination Reason</oxd-text>
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmPimPlugin/pages/configuration/SaveTerminationReason.vue:24]: Please use class `orangehrm-main-title` for headings.

File: symfony/client/src/orangehrmPimPlugin/pages/configuration/TerminationReason.vue [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-paper-container">
+      <div class="orangehrm-header-container">
+        <oxd-text tag="h6">Termination Reasons</oxd-text>
```
Comment (Super-Chama) [inline symfony/client/src/orangehrmPimPlugin/pages/configuration/TerminationReason.vue:25]: Please use class `orangehrm-main-title` for headings.

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/client/src/pages.ts [hunk 12]
```diff
@@ -50,6 +50,9 @@ import ViewOrganizationGeneralInformation from '@/orangehrmAdminPlugin/pages/org
 import Nationality from '@/orangehrmAdminPlugin/pages/nationality/Nationality.vue';
 import EditNationality from '@/orangehrmAdminPlugin/pages/nationality/EditNationality.vue';
 import SaveNationality from '@/orangehrmAdminPlugin/pages/nationality/SaveNationality.vue';
+import TerminationReason from '@/orangehrmPimPlugin/pages/configuration/TerminationReason.vue';
```
Comment (Super-Chama) [inline symfony/client/src/pages.ts:53]: `orangehrmPimPlugin` has index.ts files, you should import these files there. it will be automatically added here.

**Review by Super-Chama — CHANGES_REQUESTED** [review]
Please run `yarn lint` or pretteir format and reformat the code. found many lint issues.

File: symfony/plugins/orangehrmPimPlugin/Controller/SaveTerminationReasonController.php [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+use OrangeHRM\Core\Controller\AbstractVueController;
+use OrangeHRM\Core\Vue\Component;
+use OrangeHRM\Core\Vue\Prop;
+use OrangeHRM\Framework\Http\Request;
+use OrangeHRM\Core\Traits\Service\ConfigServiceTrait;
+
+class SaveTerminationReasonController extends AbstractVueController
+{
+    use ConfigServiceTrait;
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Controller/SaveTerminationReasonController.php:30]: any reason to use `ConfigServiceTrait`

File: symfony/plugins/orangehrmPimPlugin/config/routes.yaml [hunk 14]
```diff
@@ -343,3 +357,18 @@ pim_employee_qualifications:
   path: /pim/viewQualifications/empNumber/{empNumber}
   controller: OrangeHRM\Pim\Controller\EmployeeQualificationsController::handle
   methods: [ GET ]
+
+view_termination_record:
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/config/routes.yaml:361]: `view_termination_reason`
fix all below keys

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param $terminationReasonName
+     * @return bool
+     * @throws DaoException
+     */
+    public function isExistingTerminationReasonName($terminationReasonName): bool
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Dao/TerminationReasonConfigurationDao.php:151]: parameter type for `$terminationReasonName`

**Pull request merged**